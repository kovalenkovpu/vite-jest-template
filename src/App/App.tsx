import * as React from "react";

import logo from "../assets/logo.svg";
import classes from "./App.module.scss";

import { useHandleCount } from "../hooks/useHandleCount";
import { classExpression } from "@babel/types";

function App() {
  const { count, increaseCount } = useHandleCount(0);

  return (
    <div className={classes.app}>
      <header className={classes.appHeader}>
        <img src={logo} className={classes.appLogo} alt="logo" />
        <p>Hello Vite + React!</p>
        <p>
          <button
            className={classes.button}
            type="button"
            onClick={increaseCount}
          >
            count is: {count}
          </button>
        </p>
        <p>
          Edit <code>App.tsx</code> and save to test HMR updates.
        </p>
      </header>
    </div>
  );
}

export default App;
