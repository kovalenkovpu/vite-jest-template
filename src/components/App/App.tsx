import * as React from "react";

import logo from "../../assets/logo.svg";
import classes from "./App.module.scss";

import { useHandleCount } from "../../hooks/useHandleCount";
import { Header } from "../Header/Header";

const App: React.VFC = () => {
  const { count, increaseCount } = useHandleCount(0);

  return (
    <div className={classes.app}>
      <Header />
      <header className={classes.appHeader}>
        <img src={logo} className={classes.appLogo} alt="logo" />

        <p>
          <button
            className={classes.button}
            type="button"
            onClick={increaseCount}
          >
            count is: {count}
          </button>
        </p>
        <p>
          Edit <code>App.tsx</code> and save to test HMR updates.
        </p>
      </header>
    </div>
  );
};

export default App;
