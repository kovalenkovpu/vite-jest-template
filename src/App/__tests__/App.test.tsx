import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import App from "../App";

describe("App", () => {
  test("should change the text content when clicked on the button", () => {
    render(<App />);

    expect(screen.getByRole("button")).toHaveTextContent("0");

    const button = screen.getByRole("button");

    fireEvent.click(button);
    fireEvent.click(button);

    expect(screen.getByRole("button")).toHaveTextContent("2");
  });

  test("should match the snapshot", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
