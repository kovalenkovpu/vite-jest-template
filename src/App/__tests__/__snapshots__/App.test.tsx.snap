// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="appHeader"
    >
      <img
        alt="logo"
        class="appLogo"
        src="test-file-stub"
      />
      <p>
        Hello Vite + React!
      </p>
      <p>
        <button
          class="button"
          type="button"
        >
          count is: 0
        </button>
      </p>
      <p>
        Edit 
        <code>
          App.tsx
        </code>
         and save to test HMR updates.
      </p>
    </header>
  </div>
</DocumentFragment>
`;
